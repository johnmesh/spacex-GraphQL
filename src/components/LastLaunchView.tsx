import React from 'react'
import {
  useDataContext,
  useLastRocketLaunchQuery,
  LastLaunchDataDocument,
} from 'hooks/useDataProvider'
import { Launch } from '../types/dataTypes'
import { useQuery } from '@apollo/client'
import { GeneralDetails } from './shared/GeneralDetails'
import { RocketDetails } from './shared/RocketDetails'
import { LaunchSiteDetails } from './shared/LaunchSiteDetails'
import { LaunchLinks } from './shared/Links'
import { ShipsListHeader } from './shared/ShipListHeader'
import { ShipListBody } from './shared/ShipListBody'

export default function LastLaunchView() {
  const { launchLatest } = useDataContext()

  return (
    <div className="p-2 w-3/5 m-auto">
      <div className="flex mt-10 gap-8 h-fit h-96">
        <GeneralDetails
          mission_name={launchLatest?.mission_name}
          launch_date_local={launchLatest?.launch_date_local}
          launch_year={launchLatest?.launch_year}
          static_fire_date_utc={launchLatest?.static_fire_date_utc}
          tentative_max_precision={launchLatest?.tentative_max_precision}
          launch_success={launchLatest?.launch_success}
          details={launchLatest?.details}
        />
        <RocketDetails {...launchLatest?.rocket?.rocket} />
      </div>

      <div className="flex justify-between mt-10 gap-8 mt-28 h-fit">
        <LaunchSiteDetails {...launchLatest?.launch_site} />
        <LaunchLinks {...launchLatest?.links} />
      </div>
      {launchLatest?.ships.length > 0 && (
        <div className="mt-16 w-full drop-shadow-lg">
          <ShipsListHeader />
          <ShipListBody ships={launchLatest?.ships} />
        </div>
      )}
    </div>
  )
}
