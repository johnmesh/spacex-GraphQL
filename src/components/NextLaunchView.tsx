import React from 'react'
import { useDataContext } from 'hooks/useDataProvider'
import { RocketDetails } from './shared/RocketDetails'
import { LaunchSiteDetails } from './shared/LaunchSiteDetails'
import { LaunchLinks } from './shared/Links'
import { ShipsListHeader } from './shared/ShipListHeader'
import { ShipListBody } from './shared/ShipListBody'
import { GeneralDetails } from './shared/GeneralDetails'

export default function LastLaunchView() {
  const { loading, launchNext } = useDataContext()
  console.log('launchNext==>', launchNext)
  return (
    <div className="p-2 w-3/5 m-auto">
      <div className="flex mt-10 gap-8 h-fit h-96">
        <GeneralDetails
          mission_name={launchNext?.mission_name}
          launch_date_local={launchNext?.launch_date_local}
          launch_year={launchNext?.launch_year}
          static_fire_date_utc={launchNext?.static_fire_date_utc}
          tentative_max_precision={launchNext?.tentative_max_precision}
          launch_success={launchNext?.launch_success}
          details={launchNext?.details}
        />
        <RocketDetails {...launchNext?.rocket?.rocket} />
      </div>

      <div className="flex justify-between mt-10 gap-8 mt-28 h-fit">
        <LaunchSiteDetails {...launchNext?.launch_site} />
        <LaunchLinks {...launchNext?.links} />
      </div>
      {launchNext?.ships.length > 0 && (
        <div className="mt-16 w-full drop-shadow-lg">
          <ShipsListHeader />
          <ShipListBody ships={launchNext?.ships} />
        </div>
      )}
    </div>
  )
}
