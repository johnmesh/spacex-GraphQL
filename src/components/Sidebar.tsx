import { CaretIcon } from 'components/shared/CaretIcon'
import { LaunchpadIcon } from 'components/shared/LaunchpadIcon'
import { LoibonIcon } from 'components/shared/LoibonIcon'
import { MissionsIcon } from 'components/shared/MissionsIcon'
import { SidebarLink } from 'components/shared/SidebarLink'
import { TokenIcon } from 'components/shared/TokenIcon'
import { TreasuryIcon } from 'components/shared/TreasuryIcon'

export const Sidebar = () => {
  return (
    <div className="flex flex-col bg-lavender fixed h-full pb-8 z-40 pr-14">
      <ul className="pt-4 mx-8 pl-6">
        <li>
          <SidebarLink
            Icon={TreasuryIcon}
            text="Dashboard"
            href="/dashboard"
            className="py-4"
          />
        </li>
        <li>
          <SidebarLink
            Icon={MissionsIcon}
            text="Missions"
            href="/missions"
            className="py-4"
          />
        </li>
        <li>
          <SidebarLink
            Icon={LaunchpadIcon}
            text="Launchpad"
            href="/launchpad"
            className="py-4"
          />
        </li>
        <li>
          <SidebarLink
            Icon={LoibonIcon}
            text="Loibons"
            href="/loibons"
            className="py-4"
          />
        </li>

        <li>
          <SidebarLink
            Icon={TokenIcon}
            text="Stake Loi"
            href="#"
            className="py-4 items-center"
            isDisabled
          />
        </li>
      </ul>
      <div className="flex flex-col font-semibold text-lg mt-auto mb-14 ml-6">
        <SidebarLink
          Icon={CaretIcon}
          text="Resources"
          href="#"
          className="py-2"
        />
        <SidebarLink
          Icon={CaretIcon}
          text="Socials"
          href="#"
          className="py-4"
        />
      </div>
    </div>
  )
}
