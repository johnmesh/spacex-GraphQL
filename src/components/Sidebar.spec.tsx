import { RouterContext } from 'next/dist/shared/lib/router-context'
import { Sidebar } from 'components/Sidebar'
import { createMockrouter } from 'test-utils/create-mock-router'
import { render } from '@testing-library/react'

describe('Sidebar', () => {
  it('should render the sidebar', () => {
    const { asFragment } = render(
      <RouterContext.Provider
        value={createMockrouter({
          asPath: '/test-path',
        })}
      >
        <Sidebar />
      </RouterContext.Provider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
