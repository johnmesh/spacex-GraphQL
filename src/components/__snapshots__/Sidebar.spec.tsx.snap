// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should render the sidebar 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col bg-lavender fixed h-full pb-8 z-40 pr-6"
  >
    <ul
      class="pt-8 mx-8"
    >
      <li>
        <div
          class="flex font-semibold text-lg cursor-pointer hover:text-brown py-6"
          href="/missions"
        >
          <i
            class="w-6 h-6 mx-4"
            data-testid="missions-icon"
          >
            <test-file-stub
              classname="fill-current"
            />
          </i>
          <a
            href="/missions"
          >
            Missions
          </a>
        </div>
      </li>
      <li>
        <div
          class="flex font-semibold text-lg cursor-pointer hover:text-brown py-6"
          href="/launchpad"
        >
          <i
            class="w-6 h-6 mx-4"
            data-testid="launchpad-icon"
          >
            <test-file-stub
              classname="fill-current"
            />
          </i>
          <a
            href="/launchpad"
          >
            Launchpad
          </a>
        </div>
      </li>
      <li>
        <div
          class="flex font-semibold text-lg cursor-pointer hover:text-brown py-6"
          href="/loibons"
        >
          <i
            class="w-6 h-6 mx-4"
            data-testid="loibon-icon"
          >
            <test-file-stub
              classname="fill-current"
            />
          </i>
          <a
            href="/loibons"
          >
            Loibons
          </a>
        </div>
      </li>
      <li>
        <div
          class="flex font-semibold text-lg cursor-pointer hover:text-brown py-6"
          href="/treasury"
        >
          <i
            class="w-6 h-6 mx-4"
            data-testid="treasury-icon"
          >
            <test-file-stub
              classname="fill-current"
            />
          </i>
          <a
            href="/treasury"
          >
            Treasury
          </a>
        </div>
      </li>
      <li>
        <div
          class="flex font-semibold text-lg cursor-pointer hover:text-brown py-6 items-center"
          href="/token"
        >
          <i
            class="w-5 h-5 ml-4 mr-5"
            data-testid="token-icon"
          >
            <test-file-stub
              classname="fill-current"
            />
          </i>
          <a
            href="/token"
          >
            Token
          </a>
        </div>
      </li>
    </ul>
    <div
      class="flex flex-col font-semibold text-lg mt-auto mb-14 ml-6"
    >
      <div
        class="flex font-semibold text-lg cursor-pointer hover:text-brown py-2"
        href="/test-path#"
      >
        <i
          class="flex items-center mx-4"
          data-testid="caret-icon"
        >
          <test-file-stub
            classname="fill-current"
          />
        </i>
        <a
          href="#"
        >
          Resources
        </a>
      </div>
      <div
        class="flex font-semibold text-lg cursor-pointer hover:text-brown py-4"
        href="/test-path#"
      >
        <i
          class="flex items-center mx-4"
          data-testid="caret-icon"
        >
          <test-file-stub
            classname="fill-current"
          />
        </i>
        <a
          href="#"
        >
          Socials
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
